Marlin/src/lcd/language/language_vn.h

```diff
@@ -0,0 +1,1506 @@
+  #define MSG_Z_FADE_HEIGHT                   _UxGT("Chiều cao mờ dần")
+#endif
+#ifndef MSG_SET_HOME_OFFSETS
+// Set home offsets
+  #define MSG_SET_HOME_OFFSETS                _UxGT("Đặt bù đắp nhà")
+#endif
+#ifndef MSG_HOME_OFFSETS_APPLIED
+// Offsets applied
+  #define MSG_HOME_OFFSETS_APPLIED            _UxGT("Bù đắp được áp dụng")
+#endif
+#ifndef MSG_SET_ORIGIN
+// Set origin
+  #define MSG_SET_ORIGIN                      _UxGT("Đặt nguồn gốc")
+#endif
+#ifndef MSG_PREHEAT_1
+// Preheat
+  #define MSG_PREHEAT_1                       _UxGT("Làm nóng " PREHEAT_1_LABEL " trước")
+#endif
+#ifndef MSG_PREHEAT_1_N
+  #define MSG_PREHEAT_1_N                     MSG_PREHEAT_1 _UxGT(" ")
+#endif
+#ifndef MSG_PREHEAT_1_ALL
+// all
+  #define MSG_PREHEAT_1_ALL                   MSG_PREHEAT_1 _UxGT(" Tất cả")
+#endif
+#ifndef MSG_PREHEAT_1_END
+  #define MSG_PREHEAT_1_END                   MSG_PREHEAT_1 _UxGT(" Đầu")
+#endif
+#ifndef MSG_PREHEAT_1_BEDONLY
+// bed -- using vietnamese term for 'table' instead
+  #define MSG_PREHEAT_1_BEDONLY               MSG_PREHEAT_1 _UxGT(" Bàn")
+#endif
+#ifndef MSG_PREHEAT_1_SETTINGS
+// conf
+  #define MSG_PREHEAT_1_SETTINGS              MSG_PREHEAT_1 _UxGT(" Cấu hình")
+#endif
+#ifndef MSG_PREHEAT_2
+  #define MSG_PREHEAT_2                       _UxGT("Làm nóng " PREHEAT_2_LABEL " trước")
+#endif
+#ifndef MSG_PREHEAT_2_N
+  #define MSG_PREHEAT_2_N                     MSG_PREHEAT_2 _UxGT(" ")
+#endif
+#ifndef MSG_PREHEAT_2_ALL
+  #define MSG_PREHEAT_2_ALL                   MSG_PREHEAT_2 _UxGT(" Tất cả")
+#endif
+#ifndef MSG_PREHEAT_2_END
+  #define MSG_PREHEAT_2_END                   MSG_PREHEAT_2 _UxGT(" Đầu")
+#endif
+#ifndef MSG_PREHEAT_2_BEDONLY
+// bed -- using vietnamese term for 'table' instead
+  #define MSG_PREHEAT_2_BEDONLY               MSG_PREHEAT_2 _UxGT(" Bàn")
+#endif
+#ifndef MSG_PREHEAT_2_SETTINGS
+  #define MSG_PREHEAT_2_SETTINGS              MSG_PREHEAT_2 _UxGT(" Cấu hình")
+#endif
+#ifndef MSG_PREHEAT_CUSTOM
+  #define MSG_PREHEAT_CUSTOM                  _UxGT("Làm nóng trước tùy chỉnh")
+#endif
+#ifndef MSG_COOLDOWN
+// Cooldown
+  #define MSG_COOLDOWN                        _UxGT("Nguội xuống")
+#endif
+#ifndef MSG_SWITCH_PS_ON
+// Switch power on
+  #define MSG_SWITCH_PS_ON                    _UxGT("Bật nguồn")
+#endif
+#ifndef MSG_SWITCH_PS_OFF
+// Switch power off
+  #define MSG_SWITCH_PS_OFF                   _UxGT("Tắt nguồn")
+#endif
+#ifndef MSG_EXTRUDE
+// Extrude
+  #define MSG_EXTRUDE                         _UxGT("Ép đùn")
+#endif
+#ifndef MSG_RETRACT
+// Retract
+  #define MSG_RETRACT                         _UxGT("Rút lại")
+#endif
+#ifndef MSG_MOVE_AXIS
+// Move axis
+  #define MSG_MOVE_AXIS                       _UxGT("Di chuyển trục")
+#endif
+#ifndef MSG_BED_LEVELING
+// Bed Leveling
+  #define MSG_BED_LEVELING                    _UxGT("San Lấp Bàn")
+#endif
+#ifndef MSG_LEVEL_BED
+// Level bed
+  #define MSG_LEVEL_BED                       _UxGT("Làm bằng mặt bàn")
```

Comment:
Bed level can be translate to: " Cân chỉnh bàn in "

Reply:
I think 'Chỉnh bàn' would be good although since 'chỉnh' means adjust, it is a bit vague. 
Làm bàn bằng (with 'mặt' omitted) could work.

Reply:
Also, I use 'chỉnh' for 'tune' in other places